File: backend/internal/db/models/skill.go
package models

import "time"

type Skill struct {
	ID             string    `json:"id"`
	UserID         string    `json:"userId"`
	Name           string    `json:"name"`
	Description    string    `json:"description"`
	TriggerPattern string    `json:"triggerPattern"`
	ExecutionType  string    `json:"executionType"`
	ConfigJSON     string    `json:"config,omitempty"`
	CreatedAt      time.Time `json:"createdAt"`
}

File: frontend/src/types/skill.ts
export interface SkillConfig {
  template?: string;
  command?: string;
  args?: string[];
  [key: string]: any;
}

export interface Skill {
  id: string;
  name: string;
  description: string;
  triggerPattern?: string;
  executionType?: string;
  config?: SkillConfig;
  author?: string;
  date?: string;
  custom?: boolean;
}
